src/producer.ts

```diff
@@ -66,11 +66,17 @@ export class Producer {
                 octaveChange: octaveChange ? parseInt(octaveChange) : undefined,
               });
           }
-          const fifths = nodeElem.getElementsByTagName('fifth').item(0)?.textContent;
-          if (fifths) message.key = parseInt(fifths);
-          const beats = nodeElem.getElementsByTagName('beats').item(0)?.textContent;
-          const beatType = nodeElem.getElementsByTagName('beat-type').item(0)?.textContent;
-          if (beats && beatType) message.time = `${beats}/${beatType}`;
+          const keyElems = nodeElem.getElementsByTagName('key');
+          for (let i = 0; i < keyElems.length; i++) {
+            const fifths = parseInt(keyElems.item(i)!.getElementsByTagName('fifths').item(0)?.textContent ?? '0');
+            message.keys.push({ fifths });
```

Comment:
You need to also include the staff number in the message.

Reply:
I was too lazy as it was not needed for basic functionality, :). Done.